Userland/Libraries/LibJS/Heap/Heap.cpp

```diff
@@ -247,6 +251,8 @@ void Heap::sweep_dead_cells(bool print_report, const Core::ElapsedTimer& measure
     for (auto& weak_container : m_weak_containers)
         weak_container.remove_swept_cells({}, swept_cells);
 
+    m_inverse_roots.clear();
```

Comment:
I have a gut feeling this should happen at the end of `mark_live_cells()` instead - this function should be concerned with sweeping cells that *previously* have been marked, no need to keep the state around any longer than that.